Library/Homebrew/Gemfile

```diff
@@ -12,51 +14,68 @@ end
 # * nokogiri - use rexml instead for XML parsing
 
 # installed gems (should all be require: false)
-gem "bootsnap", require: false
-gem "byebug", require: false
-gem "json_schemer", require: false
-gem "minitest", require: false
-gem "parallel_tests", require: false
-gem "pry", require: false
-gem "ronn", require: false
-gem "rspec", require: false
-gem "rspec-github", require: false
-gem "rspec-its", require: false
-gem "rspec_junit_formatter", require: false
-gem "rspec-retry", require: false
-gem "rspec-sorbet", require: false
-gem "rubocop", require: false
-gem "rubocop-ast", require: false
-gem "simplecov", require: false
-gem "simplecov-cobertura", require: false
-gem "warning", require: false
-
-group :sorbet, optional: true do
-  gem "parlour", require: false
-  gem "sorbet-static-and-runtime", require: false
-  gem "spoom", require: false
-  gem "tapioca", require: false
+# ALL of these gems that are not vendored should be in a group
+group :ast, optional: true do
+  gem "rubocop-ast", require: false
+end
+group :bootsnap, optional: true do
+  gem "bootsnap", require: false
+end
+group :formula_test, optional: true do
+  gem "minitest", require: false
+end
+group :livecheck, optional: true do
+  gem "ruby-progressbar", require: false
+end
+group :man, optional: true do
+  gem "ronn", require: false
+end
+group :pr_publish, optional: true do
+  gem "json_schemer", require: false
 end
-
 group :prof, optional: true do
   # NOTE: ruby-prof v1.4.3 is the last version that supports Ruby 2.6.x
   # TODO: remove explicit version when HOMEBREW_REQUIRED_RUBY_VERSION >= 2.7
   gem "ruby-prof", "1.4.3", require: false
   gem "stackprof", require: false
 end
+group :pry, optional: true do
+  gem "pry", require: false
+end
+group :rspec, optional: true do
+  gem "byebug", require: false
+  gem "parallel_tests", require: false
+  gem "rspec", require: false
+  gem "rspec-github", require: false
+  gem "rspec-its", require: false
+  gem "rspec_junit_formatter", require: false
+  gem "rspec-retry", require: false
+  gem "rspec-sorbet", require: false
+  gem "simplecov", require: false
+  gem "simplecov-cobertura", require: false
+end
+group :sorbet, optional: true do
+  gem "parlour", require: false
+  gem "sorbet-static-and-runtime", require: false
+  gem "spoom", require: false
+  gem "tapioca", require: false
+end
+group :style, optional: true do
+  gem "rubocop", require: false
+  gem "rubocop-performance", require: false
+  gem "rubocop-rails", require: false
+  gem "rubocop-rspec", require: false
+  gem "rubocop-sorbet", require: false
+end
 
-# vendored gems
+# vendored gems (no group)
 gem "activesupport"
 gem "addressable"
-gem "concurrent-ruby"
 gem "patchelf"
 gem "plist"
-gem "rubocop-performance"
-gem "rubocop-rails"
-gem "rubocop-rspec"
-gem "rubocop-sorbet"
```

Comment:
We should continue to vendor these so that they are there for e.g. VSCode by default.

Reply:
I'm not sure I follow how that works. Does it not need the base rubocop? Do we need to vendor everything in those dependency trees including Rack? What happens if they're not installed (i.e. `brew style` has never been run)?

Reply:
As far as I can gather is it is [configured](https://github.com/Homebrew/brew/blob/51889e7a089f5aad0c3c4f21f0dbd3fc382f0e08/.vscode/settings.json#L10) to run https://github.com/Homebrew/brew/blob/master/Library/Homebrew/shims/gems/rubocop which runs https://github.com/Homebrew/brew/blob/master/Library/Homebrew/dev-cmd/rubocop.sh which runs `brew install-bundler-gems` anyway?

Reply:
@Bo98 Maybe not needed any more after all! If the `rubocop` shim works: that's enough.